packages/run-protocol/test/vaultFactory/test-interest.js

```diff
@@ -460,17 +400,51 @@ test('basic charge reasonable numbers monthly', async t => {
     calculator.calculateReportingPeriod(debtStatus, START_TIME + HALF_YEAR),
     {
       latestInterestUpdate: START_TIME + HALF_YEAR,
-      interest: AmountMath.make(brand, 122601n),
-      newDebt: AmountMath.make(brand, 10122601n),
+      interest: 122601n,
+      newDebt: 10122601n,
     },
   );
   // compounding: 360 days * 677 = 243720 < 246705
   t.deepEqual(
     calculator.calculateReportingPeriod(debtStatus, START_TIME + ONE_YEAR),
     {
       latestInterestUpdate: START_TIME + ONE_YEAR,
-      interest: AmountMath.make(brand, 246705n),
-      newDebt: AmountMath.make(brand, 10246705n),
+      interest: 246705n,
+      newDebt: 10246705n,
     },
   );
 });
+
+test('calculateCompoundedInterest on zero debt', t => {
+  const brand = Far('brand');
+  t.throws(() =>
+    calculateCompoundedInterest(makeRatio(0n, brand, 1n, brand), 0n, 100n),
+  );
+});
+
+// -illions
+const M = 1_000_000n;
+const B = 1_000_000_000n;
+const Q = B * B;
+
+test('calculateCompoundedInterest', t => {
+  const brand = Far('brand');
+  const cases = [
+    [1n, 1n, 1n, 1n, 1n, 1n], // no charge
```

Comment:
Isn't this a bug? Shouldn't something complain if the debt doesn't grow at all?

Reply:
I lost the context for this but it's gone anyway.